src/main/java/ch/jalu/typeresolver/FieldUtils.java

```diff
@@ -0,0 +1,142 @@
+     * @return all fields matching the filter, with the top-most parent's fields first, and this class's fields last
+     */
+    public static List<Field> getFieldsIncludingParents(Class<?> clazz, Predicate<Field> fieldFilter) {
+        return getFieldsIncludingParents(clazz, fieldFilter, true);
+    }
+
+    /**
+     * Returns all fields from the given class and its parents, recursively, that match the provided filter. Depending
+     * on the parameter, fields are either returned from top-to-bottom or bottom-to-top relative to the class's
+     * hierarchy.
+     *
+     * @param clazz the class whose fields (incl. its parents' fields) should be returned that match the filter
+     * @param fieldFilter the condition a field must fulfill in order to be part of the result
+     * @param topParentFirst true if the top-most parent's fields should come first, false for last
+     * @return all fields matching the filter, in the specified order
+     */
+    public static List<Field> getFieldsIncludingParents(Class<?> clazz, Predicate<Field> fieldFilter,
+                                                        boolean topParentFirst) {
+        LinkedList<Class<?>> classes = new LinkedList<>();
+        collectParents(clazz, (topParentFirst ? classes::addFirst : classes::addLast));
+
+        return classes.stream()
+            .flatMap(clz -> Arrays.stream(clz.getDeclaredFields()))
+            .filter(fieldFilter)
+            .collect(Collectors.toList());
+    }
+
+    /**
+     * Returns all non-synthetic, non-static fields of the given class, including all fields from superclasses.
+     *
+     * @param clazz the class whose instance fields should be retrieved
+     * @return all non-synthetic instance fields
+     */
+    public static List<Field> getRegularInstanceFieldsFromClassAndParents(Class<?> clazz) {
+        return getFieldsIncludingParents(clazz, FieldUtils::isRegularInstanceField, true);
+    }
+
+    /**
+     * Returns an optional with a field on the given class having the given name, or an empty optional if the name
+     * did not match any field. This method does not check the fields of the class's superclasses.
+     *
+     * @param clazz the class to check
+     * @param name the name of the field to resolve
+     * @return optional with the field in the class, or empty optional if there is no match
+     */
+    public static Optional<Field> tryFindField(Class<?> clazz, String name) {
+        try {
+            return Optional.of(clazz.getDeclaredField(name));
+        } catch (NoSuchFieldException ignore) {
+            return Optional.empty();
+        }
+    }
+
+    /**
+     * Returns an optional with a field of the given name on the class or one of its superclasses. If multiple fields
+     * match the given name (on different superclasses), the field on the class that is lowest in the hierarchy is
+     * returned. An empty optional is returned if no field was matched.
```

Comment:
Should probably be a boolean param / overload, given that I might want the top-most field instead? Though I guess the lowest field _is_ the most specific one... If you want another / a more specific one, probably go through `List<Field>` and filter yourself. Maybe would make sense to have a method that returns `Stream<Field>` instead of a List?